xclim/ensembles.py

```diff
@@ -190,58 +196,78 @@ def ensemble_percentiles(
     """
 
-    ds_out = xr.Dataset(attrs=ens.attrs)
-    for v in ens.data_vars:
-        # Percentile calculation forbids any chunks along realization
-        if len(ens.chunks.get("realization", [])) > 1:
-            if keep_chunk_size is None:
-                # Enable smart rechunking is chunksize exceed 2E8 elements after merging along realization
-                keep_chunk_size = (
-                    np.prod(ens[v].isel(realization=0).data.chunksize)
-                    * ens.realization.size
-                    > 2e8
-                )
-            if keep_chunk_size:
-                # Smart rechunk on dimension where chunks are the largest
-                chkDim, chks = max(
-                    ens.chunks.items(),
-                    key=lambda kv: 0 if kv[0] == "realization" else max(kv[1]),
+    if isinstance(ens, xr.Dataset):
+        out = xr.merge(
+            [
+                ensemble_percentiles(
+                    da, values, keep_chunk_size=keep_chunk_size, split=split
                 )
-                var = ens[v].chunk(
-                    {"realization": -1, chkDim: len(chks) * ens.realization.size}
-                )
-            else:
-                var = ens[v].chunk({"realization": -1})
+                for da in ens.data_vars.values()
+                if "realization" in da.dims
+            ]
+        )
+        out.attrs["history"] = update_history(
+            f"Computation of the percentiles on {ens.realization.size} ensemble members.",
+            ens,
+        )
+
+        return out
+
+    # Percentile calculation forbids any chunks along realization
+    if ens.chunks and len(ens.chunks[ens.get_axis_num("realization")]) > 1:
+        if keep_chunk_size is None:
+            # Enable smart rechunking is chunksize exceed 2E8 elements after merging along realization
+            keep_chunk_size = (
+                np.prod(ens.isel(realization=0).data.chunksize) * ens.realization.size
+                > 2e8
+            )
+        if keep_chunk_size:
+            # Smart rechunk on dimension where chunks are the largest
+            chkDim, chks = max(
+                enumerate(ens.chunks),
+                key=lambda kv: 0
+                if kv[0] == ens.get_axis_num("realization")
+                else max(kv[1]),
+            )
+            ens = ens.chunk(
+                {"realization": -1, ens.dims[chkDim]: len(chks) * ens.realization.size}
+            )
         else:
-            var = ens[v]
-
-        for p in values:
-            perc = xr.apply_ufunc(
-                _calc_perc,
-                var,
-                input_core_dims=[["realization"]],
-                output_core_dims=[[]],
-                keep_attrs=True,
-                kwargs=dict(p=p),
-                dask="parallelized",
-                output_dtypes=[ens[v].dtype],
+            ens = ens.chunk({"realization": -1})
+
+    percs = []
+    for p in values:
```

Comment:
line 239 : Would simply calling _calc_perc on all input values at once work?  i.e. instead of looping.  numpy.percentile / nanpercentile will accept tuples or list I believe.  
It might be faster as percentile calculation will do some sorting function multiple time instead of once??
We can split if desired but would also avoid the concat steps etc etc

e..g
replace kwargs with all input vals??
```
kwargs=dict(p=values),

```